### DISCUSSION TITLE
Any news on aliasing support?

### DISCUSSION THREAD

**Discussion by Ticki84** [diesel-rs/diesel#2995] [Q&A]
Hi,

I'm currently trying to make inner_join() queries over the same table to optimize (a lots) a project, so I would love to have the aliasing support. My queries are quite complex since I've implemented a nested filtering system and I've had literally no choice but making several complex queries (which are costly and could easily cause stack overflow...) whereas I could simply do all that thing with one query with the help of aliasing.

I've actually find [this PR](https://github.com/diesel-rs/diesel/pull/2254) which is quite outdated (1200+ commits behind) and doesn't provide any support in various features I also need (such as numeric).
Is there any news, update planned or alternative I could use without the need of overwriting my whole project?

Thanks for your work on the ORM by the way, it's easy and neat!

**Comment by weiznich** [comment]
@Ticki84 The statement there remains valid. Nobody has found the time yet to put in the effort required to actually implement the necessary changes. So there is no news on this. At least I do not have any concrete plan to work on this soon. Contributions for this feature are definitively welcome. 

That all written: It is possible to use a modified version of the linked PR as third party crate code to implement aliasing outside of diesel. That's what I'm using for my private projects and it works for well for my usecases. As the code is quite ad-hoc written and I do not have the capacity to clean it up and maintain yet another crate this is not published anywhere.

**Comment by Ticki84** [comment]
@weiznich Is there any way you kindly share your third party crate (as it is) then? It would save me lots of efforts to figure out how to extract it out of the box.

**Reply by weiznich** [reply]
Sharing that crate is unfortunately not possible, as it has dependencies on other internal code + it does contain many other things than the aliasing implementation. That written: The aliasing code in that crate is not that interesting at all as it is basically the same code as in the PR + some minor modifications to make it work in a third party crate.
I can offer cleaning up and publishing that code as third party crate based as paid consultant work if you are really dependent on it and don't want to do the work on your own.
